### PR TITLE
Validate vector json schema

### PR DESCRIPTION
**Pull request by xyj-3**
Issue #6

### CODE REVIEW

File: .github/workflows/validate-vector.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# SPDX-FileCopyrightText: © 2024 Kevin Lu
+# SPDX-Licence-Identifier: AGPL-3.0-or-later
+name: Validate vector.json
+on:
+  push:
+    branches: [master]
+    paths:
+      - .github/workflows/validate-vector.yaml
+      - src/validate.py
+      - data/**
```
Comment (kevinlul) [inline .github/workflows/validate-vector.yaml:10]: Can you add additional triggers on
- pull_request to master changing the same files, so we can validate this workflow within this pull request
- workflow_dispatch

File: src/validate.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# SPDX-FileCopyrightText: © 2024 Kevin Lu
```
Comment (kevinlul) [inline src/validate.py:1]: This file definitely should be yours :)

File: src/vector.schema.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# SPDX-Licence-Identifier: AGPL-3.0-or-later
+$schema: https://json-schema.org/draft/2020-12/schema
+$id: https://github.com/DawnbrandBots/yaml-yugi-limit-regulation/raw/master/src/vector.schema.yaml
+title: YAML Yugi limit regulation vectors
+description: Yu-Gi-Oh! limit regulations, stored as a mapping from Konami ID to the number allowed per deck
+type: object
+properties:
+  date:
+    description: When this limit regulation became effective
+    type: string
```
Comment (kevinlul) [inline src/vector.schema.yaml:11]: There should be a type for ISO date strings.

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: src/validate.py (hunk 2 above)
Comment (xyj-3) [reply src/validate.py:1]: The blame is good enough for me. Transferring the copyright to you.

**Comment by xyj-3** [comment]
Updated
I made action-validator stop failing due to not checked out globs being named in validate-vector action. I can undo it if you actually intended for action-validator to fail every time, or change it to exclude checking that file.

**Comment by kevinlul** [comment]
Revert that, action_validator is flawed as it doesn't parse globs like GitHub actually does. It's only for reference.

**Comment by xyj-3** [comment]
Updated

**Review by kevinlul — APPROVED** [review]
Thanks for helping with this.

**Pull request merged**